p2p/protocol/identify: tests fail with go1.24

```
2025-02-06T16:01:56.108+0100	ERROR	net/identify	identify/id.go:931	12D3KooWKA7tuYB2HpxnSTQnQZ38BCBo47uEHgYFVgix6hrEXzTy received key for remote peer 12D3KooWRyKVxttvw3x1vufF38kZRC3CZgDTeFGN4hha6aL8mSPS mismatch: 12D3KooWMNguMiS45byH1KSS3BLuRmSHspPAAyYGB8nFBPofen2G
Signed record is &{0xc00011aba0 [3 1] [10 38 0 36 8 1 18 32 160 14 5 250 194 225 122 29 245 210 173 158 109 0 177 18 190 242 118 11 110 60 88 111 70 21 14 76 169 5 221 35 16 221 196 157 245 180 211 233 144 24 26 10 10 8 4 127 0 0 1 6 135 175 26 13 10 11 4 127 0 0 1 145 2 153 39 205 3] [136 206 179 191 35 3 208 203 129 71 207 227 28 114 225 185 128 8 171 145 115 240 246 221 140 94 8 12 113 31 195 93 75 161 13 111 231 124 159 196 119 84 255 47 59 126 39 227 54 122 245 221 173 224 0 197 233 14 133 212 86 147 125 3] 0xc00062d860 <nil> {{{} 1} {0 0}}}
Done sending msg
2025-02-06T16:01:57.113+0100	ERROR	net/identify	identify/id.go:801	error getting peer record from Identify message: failed to validate envelope: invalid signature or incorrect domain
--- FAIL: TestIDService (0.01s)
    --- FAIL: TestIDService/withObsAddrManager=false (0.01s)
        id_test.go:177: test peer1 has peer2 addrs correctly
        id_test.go:48: agent version mismatch <nil>
        id_test.go:189: test peer2 has peer1 addrs correctly
        id_test.go:48: agent version mismatch <nil>
        id_test.go:203: testing addrs just after disconnect
        id_test.go:213: testing addrs after TTL expiration
    --- FAIL: TestIDService/withObsAddrManager=true (0.01s)
        id_test.go:177: test peer1 has peer2 addrs correctly
        id_test.go:48: agent version mismatch <nil>
        id_test.go:189: test peer2 has peer1 addrs correctly
        id_test.go:48: agent version mismatch <nil>
        id_test.go:203: testing addrs just after disconnect
        id_test.go:213: testing addrs after TTL expiration
--- FAIL: TestLargeIdentifyMessage (0.01s)
    id_test.go:682: test peer1 has peer2 addrs correctly
    id_test.go:48: agent version mismatch <nil>
    id_test.go:696: test peer2 has peer1 addrs correctly
    id_test.go:48: agent version mismatch <nil>
    id_test.go:710: testing addrs just after disconnect
    id_test.go:720: testing addrs after TTL expiration
FAIL
FAIL	github.com/libp2p/go-libp2p/p2p/protocol/identify	12.124s
FAIL
```
[golang/go#d79e6bec6389dfeeec84a64f283055090615bad1](https://github.com/golang/go/commit/d79e6bec6389dfeeec84a64f283055090615bad1) is the first bad commit

--- a/p2p/protocol/identify/id.go
+++ b/p2p/protocol/identify/id.go
@@ -19,6 +19,7 @@ import (
 	"github.com/libp2p/go-libp2p/core/protocol"
 	"github.com/libp2p/go-libp2p/core/record"
 	"github.com/libp2p/go-libp2p/p2p/host/eventbus"
+	useragent "github.com/libp2p/go-libp2p/p2p/protocol/identify/internal/user-agent"
 	"github.com/libp2p/go-libp2p/p2p/protocol/identify/pb"
 
 	logging "github.com/ipfs/go-log/v2"
@@ -53,8 +54,6 @@ const (
 	recentlyConnectedPeerMaxAddrs = 20
 	connectedPeerMaxAddrs         = 500
 )
-
-var defaultUserAgent = "github.com/libp2p/go-libp2p"
 
 type identifySnapshot struct {
 	seq       uint64
@@ -188,7 +187,7 @@ func NewIDService(h host.Host, opts ...Option) (*idService, error) {
 		opt(&cfg)
 	}
 
-	userAgent := defaultUserAgent
+	userAgent := useragent.DefaultUserAgent()
 	if cfg.userAgent != "" {
 		userAgent = cfg.userAgent
 	}

--- a/p2p/protocol/identify/id_test.go
+++ b/p2p/protocol/identify/id_test.go
@@ -27,6 +27,7 @@ import (
 	"github.com/libp2p/go-libp2p/p2p/net/swarm"
 	swarmt "github.com/libp2p/go-libp2p/p2p/net/swarm/testing"
 	"github.com/libp2p/go-libp2p/p2p/protocol/identify"
+	useragent "github.com/libp2p/go-libp2p/p2p/protocol/identify/internal/user-agent"
 	"github.com/libp2p/go-libp2p/p2p/protocol/identify/pb"
 
 	mockClock "github.com/benbjohnson/clock"
@@ -45,7 +46,7 @@ func testKnowsAddrs(t *testing.T, h host.Host, p peer.ID, expected []ma.Multiadd
 func testHasAgentVersion(t *testing.T, h host.Host, p peer.ID) {
 	v, err := h.Peerstore().Get(p, "AgentVersion")
 	require.NoError(t, err, "fetching agent version")
-	require.Equal(t, "github.com/libp2p/go-libp2p", v, "agent version")
+	require.Equal(t, useragent.DefaultUserAgent(), v, "agent version")
 }
 
 func testHasPublicKey(t *testing.T, h host.Host, p peer.ID, shouldBe ic.PubKey) {

--- a/p2p/protocol/identify/id_test.go
+++ b/p2p/protocol/identify/id_test.go
@@ -104,7 +104,7 @@ func emitAddrChangeEvt(t *testing.T, h host.Host) {
 	}
 }
 
-// TestIDServiceWait gives the ID service 1s to finish after dialing
+// TestIDService gives the ID service 1s to finish after dialing
 // this is because it used to be concurrent. Now, Dial wait till the
 // id service is done.
 func TestIDService(t *testing.T) {

--- a/p2p/protocol/identify/id_test.go
+++ b/p2p/protocol/identify/id_test.go
@@ -44,9 +44,8 @@ func testKnowsAddrs(t *testing.T, h host.Host, p peer.ID, expected []ma.Multiadd
 
 func testHasAgentVersion(t *testing.T, h host.Host, p peer.ID) {
 	v, err := h.Peerstore().Get(p, "AgentVersion")
-	if v.(string) != "github.com/libp2p/go-libp2p" { // this is the default user agent
-		t.Error("agent version mismatch", err)
-	}
+	require.NoError(t, err, "fetching agent version")
+	require.Equal(t, "github.com/libp2p/go-libp2p", v, "agent version")
 }
 
 func testHasPublicKey(t *testing.T, h host.Host, p peer.ID, shouldBe ic.PubKey) {

--- a/p2p/protocol/identify/internal/user-agent/user_agent.go
+++ b/p2p/protocol/identify/internal/user-agent/user_agent.go
@@ -1,9 +1,15 @@
-package identify
+package useragent
 
 import (
 	"fmt"
 	"runtime/debug"
 )
+
+func DefaultUserAgent() string {
+	return defaultUserAgent
+}
+
+var defaultUserAgent = "github.com/libp2p/go-libp2p"
 
 func init() {
 	bi, ok := debug.ReadBuildInfo()

